src/core/nonbonded_interactions/nonbonded_interaction_data.hpp

```diff
@@ -78,176 +225,63 @@ struct IA_parameters {
-  /** \name BMHTF NaCl potential */
-  /*@{*/
-  double BMHTF_A = 0.0;
-  double BMHTF_B = 0.0;
-  double BMHTF_C = 0.0;
-  double BMHTF_D = 0.0;
-  double BMHTF_sig = 0.0;
-  double BMHTF_cut = INACTIVE_CUTOFF;
-  double BMHTF_computed_shift = 0.0;
-  /*@}*/
+  BMHTF_Parameters bmhtf;
 #endif
 
 #ifdef MORSE
-  /** \name Morse potential */
-  /*@{*/
-  double MORSE_eps = INACTIVE_CUTOFF;
-  double MORSE_alpha = INACTIVE_CUTOFF;
-  double MORSE_rmin = INACTIVE_CUTOFF;
-  double MORSE_cut = INACTIVE_CUTOFF;
-  double MORSE_rest = INACTIVE_CUTOFF;
-  /*@}*/
+  Morse_Parameters morse;
 #endif
 
 #ifdef BUCKINGHAM
-  /** \name Buckingham potential */
-  /*@{*/
-  double BUCK_A = 0.0;
-  double BUCK_B = 0.0;
-  double BUCK_C = 0.0;
-  double BUCK_D = 0.0;
-  double BUCK_cut = INACTIVE_CUTOFF;
-  double BUCK_discont = 0.0;
-  double BUCK_shift = 0.0;
-  double BUCK_F1 = 0.0;
-  double BUCK_F2 = 0.0;
-  /*@}*/
+  Buckingham_Parameters buckingham;
 #endif
 
 #ifdef SOFT_SPHERE
-  /** \name soft-sphere potential */
-  /*@{*/
-  double soft_a = 0.0;
-  double soft_n = 0.0;
-  double soft_cut = INACTIVE_CUTOFF;
-  double soft_offset = 0.0;
-  /*@}*/
+  SoftSphere_Parameters soft_sphere;
 #endif
 
 #ifdef AFFINITY
-  /** \name affinity potential */
-  /*@{*/
-  int affinity_type = -1;
-  double affinity_kappa = INACTIVE_CUTOFF;
-  double affinity_r0 = INACTIVE_CUTOFF;
-  double affinity_Kon = INACTIVE_CUTOFF;
-  double affinity_Koff = INACTIVE_CUTOFF;
-  double affinity_maxBond = INACTIVE_CUTOFF;
-  double affinity_cut = INACTIVE_CUTOFF;
-  /*@}*/
+  Affinity_Parameters affinity;
 #endif
 
 #ifdef MEMBRANE_COLLISION
-  /** \name membrane collision potential */
-  /*@{*/
-  double membrane_a = 0.0;
-  double membrane_n = 0.0;
-  double membrane_cut = INACTIVE_CUTOFF;
-  double membrane_offset = 0.0;
-  /*@}*/
+  Membrane_Parameters membrane;
 #endif
 
 #ifdef HAT
-  /** \name hat potential */
-  /*@{*/
-  double HAT_Fmax = 0.0;
-  double HAT_r = INACTIVE_CUTOFF;
-  /*@}*/
+  Hat_Parameters hat;
 #endif
 
 #ifdef LJCOS
-  /** \name Lennard-Jones+Cos potential */
-  /*@{*/
-  double LJCOS_eps = 0.0;
-  double LJCOS_sig = 0.0;
-  double LJCOS_cut = INACTIVE_CUTOFF;
-  double LJCOS_offset = 0.0;
-  double LJCOS_alfa = 0.0;
-  double LJCOS_beta = 0.0;
-  double LJCOS_rmin = 0.0;
-  /*@}*/
+  LJcos_Parameters ljcos;
 #endif
 
 #ifdef LJCOS2
-  /** \name Lennard-Jones with a different Cos potential */
-  /*@{*/
-  double LJCOS2_eps = 0.0;
-  double LJCOS2_sig = 0.0;
-  double LJCOS2_cut = INACTIVE_CUTOFF;
-  double LJCOS2_offset = 0.0;
-  double LJCOS2_w = 0.0;
-  double LJCOS2_rchange = 0.0;
-  /*@}*/
+  LJcos2_Parameters ljcos2;
 #endif
 
 #ifdef GAY_BERNE
-  /** \name Gay-Berne potential */
-  /*@{*/
-  double GB_eps = 0.0;
-  double GB_sig = 0.0;
-  double GB_cut = INACTIVE_CUTOFF;
-  double GB_k1 = 0.0;
-  double GB_k2 = 0.0;
-  double GB_mu = 0.0;
-  double GB_nu = 0.0;
-  double GB_chi1 = 0.0;
-  double GB_chi2 = 0.0;
-  /*@}*/
+  GayBerne_Parameters gb;
```

Comment:
In this morning's meeting, we proposed spelling out `gb` to `gay_berne`. After that we should be good to merge.